api/openapi/my_api.yaml

```diff
@@ -446,34 +454,7 @@ components:
-                    description: "A value such as 'The value was specified by the end user.' indicates that any value in the EHR has been overriden by the client via the input values to the endpoint, and they will be simply passed back here."
-                    type: string
-                    example: "The value was specified by the end user."
-                  why:
-                    description: "Provides the client with a human-readable, non-standard justification for why the plugin needs this particular variable. This can be rendered by the client to provide the end-user with rationale for the patient values, but does not address how the value was computed."
-                    type: string
-                    example: "Age is used to calculate the creatinine clearance. Dosing is lower for geriatric patient and contraindicated for pediatric patients"
-                  variableValue:
-                    description: "The justification id maps to only one PatientVariable, which in turn, defines all the legal values (legalValues) for this id. The legalValues define the type for the variableValue, and so can be used to inform the client on how to render the variableValue. The legalValues field here is optional, so the client can find the type by calling the '/config' endpoint, matching the 'id' with that in this justification schema, and looking up 'legalValues'. The variableValue may also include the 'units' field, which is optional, and if left off, implies the system default units. NOTE: The 'variableValue' is actually an object type, but we use 'string' here to overcome a shortcoming of the swagger ui. This design choice was made in order to prioritize a better interactive documentation experience for the API user."
-                    type: object
-                    example: {"value": "0.5", "units": "years"}
-                  legalValues:
-                    description: "The 'legalValues' will always be present in the '/config' endpoint response and need not be provided to or from the client in the '/guidance' endpoint call. Use the JSON schema, https://json-schema.org/draft/2019-09/json-schema-validation.html. For example, { type: 'string', 'enum': [ 'Gentamicin', 'Amikacin' ] } to define an enumeration,  { 'type': 'number' } for a float, and { 'type': 'number', 'minimum': '0' } for a positive number, and { 'type': 'integer' } for an integer. NOTE: This value is actually an object type, but we use 'string' here to overcome a shortcoming of the swagger ui. This design choice was made in order to prioritize a better interactive documentation experience for the API user. "
-                    type: string
-                    example: {"type": "number", "minimum": "0"}
-                  timestamp:
-                    type: string
-                    format: date-time
-                    description: "If not specified, assume current time"
+                  $ref: '#/components/schemas/PatientVariableValue/properties'
```

Comment:
@rajshruti18 I am not sure whether swagger API allows to reference properties directly, so I pasted this updated my_api.yaml file into https://editor.swagger.io site and it does not pass swagger validation, and give a structural error as follows: ```Structural error at components.schemas.Guidance.properties.justification.properties.inputs.items.properties.$ref should be object```. So it looks like we can only reference objects as a whole. It looks like the only way we can consolidate the PatientVariableValue object to be used by both ```UserSuppliedPatientVariables``` and ```inputs``` in ```justification``` in guidance request response is to relax the why field from required to optional in ```inputs``` object definition in `justification` in guidance request response. We will need input from @krobasky on whether this is a good idea or not.

Reply:
I thought this would happen. Waiting for reply from @krobasky .

Reply:
Yes, I think its ok to relax the why field to optional so that the same schema can be used across other PDS APIs.

Reply:
@hyi Relaxed the requirement of why field. Please review it and let me know if I can merge it.